### PR TITLE
Fixed deprecated method usages of MessageManager

### PR DESCRIPTION
**Pull request by drpayyne**
### Description

- Fixed **all** deprecated method usages of MessageManager in `Magento_Wishlist`
- Fixed appropriate PHPUnit tests

### Manual testing scenarios

- ONLY code modifications

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminAddImageToWYSIWYGProductTest.xml [hunk 1]
```diff
@@ -75,6 +75,7 @@
         <waitForLoadingMaskToDisappear stepKey="waitForLoading13"/>
         <see selector="{{ProductShortDescriptionWYSIWYGToolbarSection.CreateFolder}}" userInput="Create Folder" stepKey="seeCreateFolderBtn2" />
         <waitForLoadingMaskToDisappear stepKey="waitForLoading14"/>
+        <wait stepKey="wait" time="3"/>
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminAddImageToWYSIWYGProductTest.xml]: Why we need it? and why it wasn't needed earlier?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @drpayyne,
Could you review and fix failing tests/adjust your code?

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminAddImageToWYSIWYGProductTest.xml (hunk 1 above)
Comment (drpayyne) [reply app/code/Magento/Catalog/Test/Mftf/Test/AdminAddImageToWYSIWYGProductTest.xml]: The test fails since there’s a delay so I added it.

**Comment by drpayyne** [comment]
@ihor-sviziev I have fixed the failing MFTF and integration test. Replaced the <wait/> with a better alternative and also fixed unnecessary code in the MFTF file.

**Review by ihor-sviziev — APPROVED** [review]

**Review by lbajsarowicz — APPROVED** [review]
Nice job! Thanks for your contribution.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6168](https://jira.corp.magento.com/browse/ENGCOM-6168) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by drpayyne** [comment]
@lbajsarowicz Thank _you_!

**Comment by engcom-Alfa** [comment]
Hi @drpayyne 

During testing, we faced the issue

_**Problem:**_ __404 Not found_ after updating Wish List

**Steps to reproduce:**
1. Add simple product to Wishlist;
2. From Wish List click on edit :pen: product button;
3. Change, for example, the quantity of product;
4. Click on "Update Wish List" button;

**Actual Result:** _404 not found_ error occurs
after

Probably, the problem is with _HttpGetActionInterface_ in _UpdateItemOptions_ class

@drpayyne Could you take a look?

Thanks!

**Comment by drpayyne** [comment]
Hi @engcom-Alfa, thank you for your detailed comment. I've fixed it and sorry I missed that out. Thank you!

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6168](https://jira.corp.magento.com/browse/ENGCOM-6168) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**